unison-share-api/src/Unison/Server/Endpoints/FuzzyFind.hs

```diff
@@ -134,22 +126,22 @@ instance ToSample FoundResult where
   toSamples _ = noSamples
 
 serveFuzzyFind ::
-  Handler () ->
-  Codebase IO Symbol Ann ->
+  forall m.
+  MonadIO m =>
+  Codebase m Symbol Ann ->
   Maybe SBH.ShortBranchHash ->
   Maybe HashQualifiedName ->
   Maybe Int ->
   Maybe Width ->
   Maybe String ->
-  Handler (APIHeaders [(FZF.Alignment, FoundResult)])
```

Comment:
I converted all of these handlers to just use the `Backend` monad rather than a full web handler so that the caller (e.g. enlil) can use them as it likes, and can recover from Backend errors more easily.

I moved the cache-control headers up a layer.